### ISSUE TITLE
Information regarding sbert weights on huggingface model hub

### ISSUE THREAD

**Issue by prajjwal1** [UKPLab/sentence-transformers#378]
Hi Nils,
`sentence-transformers` were recently added in the model hub. I wanted to know the following if :
- all these models were trained on NLI (which dataset was used ? MNLI+SNLI as the paper suggests)
- how many epochs was the model trained for?
- If I want to use it for inference on NLI task, do I need to train an additional linear layer ? So freeze the AutoModel, train a nn.Linear for 1 epoch on the mean output and then perform inference. Can you please confirm this usage ?
- Which objective was used ? I trained it on MNLI with the model frozen and training the last linear layer, it gets around 46% accuracy after epoch 1. Seems like it has not been trained on classification objective. Section 3 of the paper mentions three objective functions classification, regression and triplet. As per Fig 2. you seem to be using regression for computing similarity scores at inference. As per Fig 1. it says that you used classification objective for fine-tuning. Based on the performance I’m getting, I doubt that these weights are coming from model trained on classification. Can you please clarify the objective ?

I also opened a post on [HF forum](https://discuss.huggingface.co/t/info-regarding-sentence-transformers/808/4). Seems like no one was fully sure of it. That's why I'm asking here. Thanks in advance.

**Comment by nreimers** [comment]
Hi @prajjwal1 

Yes, the models where trained on SNLI + MNLI. However, as the objective of sentence-transformers is to produce sentence embeddings, that can be compared with cosine similarity (e.g. for clustering or semantic search), the training procedure is different.

The two sentences were passed individually to BERT, both producing vector u and v. These vectors where concatenated (u, v, |u-v|) and passed to a softmax classifier (linear layer + softmax). The softmax classifier was trained with negative log likelihood.

The models were usually trained for 1 epoch.

To learn a meaningful classifier, you would need to pass the embeddings individually to the models, perform mean pooling, concatenate (u, v, |u-v|) and train your linear layer with softmax.

How do you want to do inference on NLI task? Do you want to classify pairs? Then these models are suboptimal. In that case, it is better to pass on a single string to BERT in the format: Sentence A [SEP] Sentence B.

In this case, BERT can perform cross-attention between the two sentences. The classification accuracy will be higher.

The published models and this library here are more targeted for generating sentence embeddings, that can be compared with cosine similarity.

Best
Nils Reimers

**Comment by prajjwal1** [comment]
Thanks for the detailed answer.
